### PR TITLE
Python code improvements

### PR DESCRIPTION
**Pull request by sisp**
I've done a little Python code cleanup, split into several commits to separate the changes by type.

### CODE REVIEW

File: src/plugins/privacy/plugin.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def _parse_fragment(self, fragment: str):
 
     # Parse and extract all external assets from a media file using a preset
     # regular expression, and return all URLs found.
-    def _parse_media(self, initiator: File) -> "list[URL]":
+    def _parse_media(self, initiator: File) -> list[URL]:
```
Comment (squidfunk) [inline src/plugins/privacy/plugin.py:250]: Does this work with Python 3.8?

Comment (sisp) [reply src/plugins/privacy/plugin.py]: Oh, no, you're right, it doesn't. I've dropped that commit.

Comment (squidfunk) [reply src/plugins/privacy/plugin.py]: We'll add proper tests next year for all supported Python versions, that will safe-guard us against those cases.

Comment (sisp) [reply src/plugins/privacy/plugin.py]: How about also using a proper dependency manager like [Poetry](https://github.com/python-poetry/poetry), which supports a lock file and dev-dependencies, and [Ruff](https://github.com/astral-sh/ruff) as formatter and linter? Plus, of course, mypy for the type-checking that would have caught the above problem under Python 3.8.

Comment (squidfunk) [reply src/plugins/privacy/plugin.py]: Yes, we're on it, please see #6473. Note that my adventure into Python is kind of an accident, as I'd never would've thought that I will develop that many MkDocs plugins. The code base grew so large with the new capabilities, that we now need to professionalize the project. We will do this in 2024.

**Pull request merged**